// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package l3_attackmitigator;

import "context.proto";
import "l3_centralizedattackdetector.proto";

service L3Attackmitigator{
  rpc PerformMitigation (L3AttackmitigatorOutput) returns (l3_centralizedattackdetector.StatusMessage) {}
  rpc GetMitigation (context.Empty) returns (context.Empty) {}
  rpc GetConfiguredACLRules (context.Empty) returns (ACLRules) {}
}


message L3AttackmitigatorOutput {
	float confidence = 1;
	string timestamp = 2;
	string ip_o = 3;
	string ip_d = 4;	
	string tag_name = 5;
	int32 tag = 6;
	string flow_id = 7;
	string protocol = 8;
	string port_o = 9;
	string port_d = 10;
	string ml_id = 11;
	context.ServiceId service_id = 12;
	context.EndPointId endpoint_id = 13; 
	float time_start = 14;
	float time_end = 15;
}

message ACLRules {
	repeated context.ConfigRule acl_rules = 1;
}
